WebHostLib/api/tracker.py

```diff
@@ -153,80 +149,81 @@ class PlayerStatus(TypedDict):
             if slot_info.type != SlotType.group or not slot_info.group_members:
                 continue
-            groups_in_team.append(
+            groups.append(
                 {
+                    "team": team,
                     "slot": player,
                     "name": slot_info.name,
                     "members": list(slot_info.group_members),
                 })
-    class PlayerName(TypedDict):
-        player: int
-        name: str
 
-    player_names: list[dict[str, str | list[PlayerName]]] = []
-    """Slot names of all players."""
-    for team, players in all_players.items():
-        per_team_player_names: list[PlayerName] = []
-        team_names = {"team": team, "players": per_team_player_names}
-        player_names.append(team_names)
-        for player in players:
-            per_team_player_names.append({"player": player, "name": tracker_data.get_player_name(team, player)})
+    return {
+        "groups": groups,
+        "datapackage": tracker_data._multidata["datapackage"],
+    }
 
-    class PlayerGame(TypedDict):
-        player: int
-        game: str
+# It should be exceedingly rare that slot data is needed, so it's separated out.
+@api_endpoints.route("/slot_data_tracker/<suuid:tracker>")
+@cache.memoize(timeout=300)
+def tracker_slot_data(tracker: UUID) -> list[PlayerSlotData]:
+    """
+    Outputs json data to <root_path>/api/slot_data_tracker/<id of current session tracker>.
 
-    games: list[dict[str, int | list[PlayerGame]]] = []
-    """The game each player is playing."""
-    for team, players in all_players.items():
-        player_games: list[PlayerGame] = []
-        team_games = {"team": team, "players": player_games}
-        games.append(team_games)
-        for player in players:
-            player_games.append({"player": player, "game": tracker_data.get_player_game(team, player)})
+    :param tracker: UUID of current session tracker.
 
-    class PlayerSlotData(TypedDict):
-        player: int
-        slot_data: dict[str, Any]
+    :return: Slot data for all players in the room. Typing completely arbitrary per game.
+    """
+    room: Room | None = Room.get(tracker=tracker)
+    if not room:
+        abort(404)
+    tracker_data = TrackerData(room)
+
+    all_players: dict[int, list[int]] = tracker_data.get_all_players()
 
-    slot_data: list[dict[str, int | list[PlayerSlotData]]] = []
+    slot_data: list[PlayerSlotData] = []
     """Slot data for each player."""
     for team, players in all_players.items():
-        player_slot_data: list[PlayerSlotData] = []
-        team_slot_data = {"team": team, "players": player_slot_data}
-        slot_data.append(team_slot_data)
         for player in players:
-            player_slot_data.append({"player": player, "slot_data": tracker_data.get_slot_data(team, player)})
+            slot_data.append({"team": team, "player": player, "slot_data": tracker_data.get_slot_data(team, player)})
```

Comment:
Why does slot data have teams in it? It seems wasteful unless there is a plan for having different slot data per team

Reply:
True, I blame https://github.com/ArchipelagoMW/Archipelago/pull/5272 still not being merged as the source of this mistake, otherwise the team argument wouldn't exist to slot data in the first place.